SteamKit2/SteamKit2/Steam/Discovery/SmartCMServerList.cs

```diff
@@ -264,7 +233,8 @@ private IPEndPoint GetNextServerCandidateInternal()
                 ResetOldScores();
 
                 var serverInfo = servers
-                    .Select( (s, index) => new { EndPoint = s.EndPoint, IsBad = s.LastBadConnectionDateTimeUtc.HasValue, Index = index } )
+                    .Where( s => (s.Record.ProtocolTypes & supportedProtocolTypes) > 0 )
```

Comment:
But yeah. Not overly concerned about perf on this one. I'm more concerned about CMClient and server list.